README.md

```diff
@@ -185,9 +215,261 @@ docker run 348099934012.dkr.ecr.eu-central-1.amazonaws.com/kilt/prototype-chain
 ```
 The node should be connected to the KILT testnet.
 
-## Updating with latest substrate-node-template
+## Node Modules functionalities
+
+The KILT blockchain is the heart and soul behind KILT protocol. It provides the immutable
+transaction ledger for the various processes in the network.
+
+`Building on the Parity Substrate Blockchain Framework`
+
+During our first whiteboard phase, we were thinking about developing the KILT protocol on
+Ethereum smart-contracts, but we realised that we would have less freedom of setting
+transaction costs, while incurring a high level of overhead. Instead, we started our
+development on [Parity Substrate](https://www.parity.io/substrate/), a general blockchain framework, and built up the KILT blockchain from scratch based on its module library.
+
+Building our blockchain on Parity Substrate has multiple advantages. Substrate has a very
+good fundamental [architecture](https://docs.substrate.dev/docs/architecture-of-a-runtime) and [codebase](https://github.com/paritytech/substrate) created by blockchain experts. Substrate
+framework is developed in Rust, a memory efficient and fast compiled system programming
+language, which provides a secure environment with virtually no runtime errors. Moreover, the
+node runtime is also compiled to WebAssembly, so older version native nodes can always run
+the latest version node runtime in a WebAssembly virtual machine to bypass the problem of a
+blockchain fork. Importantly, there is a vibrant developer community and rich [documentation](https://docs.substrate.dev/).
+
+Our implementation is based on the [substrate-node-template](https://github.com/rstormsf/substrate-node-template) library (skeleton template for
+quickly building a substrate based blockchain), which is linked to the main Substrate
+codebase.
+
+Remote Procedure Calls
```

Comment:
Can you format all titles of this section as markdown titles? 
`### Remote Procedure Calls`
`### Blocktime`
...etc.

Reply:
Sure, tho I omitted that intentionally, because that part is just taken from the whitepaper and I don't want to  further overload the ToC. And for me it feels more like a text paragraph, which should be taken as a whole. Your thoughts?

Reply:
OK, valid point regarding the ToC. My concern is that this big unformatted content block might be hard to consume ==> let's simply make the titles bold.

Reply:
good suggestion